### PR TITLE
test: Add a (failing) test for removing case insensitive duplicates.

### PR DESCRIPTION
**Pull request by JeffFaer**

### CODE REVIEW

File: goldens/duplicates.out [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
+  // keep-sorted-test start remove_duplicates=yes
+  bar
+  foo
+  // keep-sorted-test end
+
+XXX: This behavior is incorrect.
+Remove duplicates, case insensitive:
+  // keep-sorted-test start remove_duplicates=yes case=no
+  bar
+  FoO
```
[the remark this answers is not in the document]
Comment (JeffFaer) [reply goldens/duplicates.out:11]: Or perhaps we move `remove_duplicates` to a pre-sorting option so that people wouldn't expect it to work with `by_regex` or `case=no`. In a similar vein, perhaps `case=no` should be a pre-sorting option as well since it applies to `group_prefixes`.

[the remark this answers is not in the document]
Comment (KatrinaHoffert) [reply goldens/duplicates.out:11]: Has anyone reported this case in the real world yet? I'm inclined to suspect it may be an edge case to not deal with mostly because I'm not sure we can even satisfactorily answer the question of which case to keep. I suspect people might be upset by any choice we would make.

I think it's actually most common to use `case=no` on data that _is_ case sensitive in reality, but the user just wants the sorting to be case insensitive, as it can be much more intuitive, especially when there's initialisms involved (e.g., `REST` sorting so far away from `Rent` can be unintuitive). `remove_duplicates` defaults to true, so we'd want to be careful simply because users are typically not taking it into account.

**Review by KatrinaHoffert — APPROVED** [review]

File: goldens/duplicates.out (hunk 1 above)
[the remark this answers is not in the document]
Comment (JeffFaer) [reply goldens/duplicates.out:11]: > I think it's actually most common to use case=no on data that is case sensitive in reality, but the user just wants the sorting to be case insensitive, as it can be much more intuitive

Ah right, that makes perfect sense! Reverting
